### PR TITLE
Xcode 14 modernization

### PR DESCRIPTION
**Pull request by ksuther**
Can you double-check my accessibility modernization in SRRecorderControl.m?

### CODE REVIEW

**Comment by vendruscolo** [comment]
It provides a worse UX, previously, it would read the current shortcut, now it only says it's a button. And it would be nice to post a notification to confirm it was recorded.

Do you want some help in sorting this out?

**Comment by ksuther** [comment]
Yeah, if you don't mind you'd probably be much faster at it :)

**Comment by vendruscolo** [comment]
I pushed some changes. There's only one thing that I can't get to work, but I feel like it's not possible:

1. Highlight the recorder
2. Start recording
3. <VO>-← to move focus to the label before it

I'd like to cancel recording, but I can't get any accessibility hook to work. The closest one is `accessibilityFocused` but it seems is never called when the control loses focus (so I only see `YES` values).

Other than that VO now reads the button as before, and it speaks the recorded shortcut.

**Comment by ksuther** [comment]
Thanks, that's still better than before. @rustle any opinions?

**Review by vendruscolo — APPROVED** [review]

File: Library/SRRecorderControl.m [hunk 1]
```diff
@@ -82,6 +82,8 @@ typedef NS_ENUM(NSUInteger, _SRRecorderControlButtonTag)
     _SRRecorderControlMainButtonTag = 2
 };
 
+@interface SRRecorderControl () <NSAccessibility, NSViewToolTipOwner>
```
Comment (rustle) [inline Library/SRRecorderControl.m:85]: It's unusual to add `NSAccessibility` conformance like this. What are you indicating?

File: Library/SRRecorderControl.m [hunk 2]
```diff
@@ -422,6 +430,11 @@ - (NSString *)accessibilityLabel
     return label;
 }
 
+- (NSString *)accessibilityTitle
```
Comment (rustle) [inline Library/SRRecorderControl.m:433]: Does this duplication resolve a specific issue?

Comment (rustle) [reply Library/SRRecorderControl.m:433]: (label is intended to be a clean replacement for title, but there have definitely been bugs historically)

File: Library/SRRecorderControl.m (hunk 1 above)
Comment (rustle) [reply Library/SRRecorderControl.m:85]: You might want `NSAccessibilityButton` which will populate the role for you

**Comment by ksuther** [comment]
I just noticed there's a bunch of upstream changes, taking a look at those and figuring out if we need any of this at all. I might have wasted a bunch of time by not looking there first, oops.

File: Library/SRRecorderControl.m (hunk 2 above)
Comment (vendruscolo) [reply Library/SRRecorderControl.m:433]: That was done in 349fa6841d88feb22eb45014bbe6dd326cb5c7a4 - IIRC VoiceOver reads the title of the button but we were only providing the label.
It results in having VO doing a better job at reading the state of the button.

**Comment by vendruscolo** [comment]
After speaking with Doug, there's no need to stop recording mode when moving focus away from the button. This happens automatically if you navigate to a different (interactive) control.

**Comment by ksuther** [comment]
I'm going to close this in favor of https://github.com/Kentzo/ShortcutRecorder/pull/150. Might as well get on the latest upstream version since they've modernized a lot in the last few years.

Sorry that I burned up a bunch of review cycles on this!

**Pull request closed without merging**